pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -139,178 +400,41 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 	}
 
 	if _, ok := data.(*getRepoDataBySearchResponse); ok {
-		for _, repo := range searchRepos {
-			var name string
-			var defaultBranch string
-			var branchCursor *string
-			var branches []BranchNode
-
-			if n, ok := repo.Node.(*SearchNodeRepository); ok {
-				name = n.Name
-				defaultBranch = n.DefaultBranchRef.Name
-			}
-
-			// Getting contributor count via the graphql api is very process heavy
-			// as you have to get all commits on the default branch and then
-			// iterate through each commit to get the author and committer, and remove
-			// duplicate values. The default branch could be thousands of commits,
-			// which would require tons of pagination and requests to the api. Doing
-			// so via the rest api is much more efficient as it's a direct endpoint
-			// with limited pagination.
-			// Due to the above, we'll only run this actual code when the metric
-			// is explicitly enabled.
-			if ghs.cfg.MetricsBuilderConfig.Metrics.GitRepositoryContributorCount.Enabled {
-				gc := github.NewClient(ghs.client)
-				contribs, _, err := gc.Repositories.ListContributors(ctx, ghs.cfg.GitHubOrg, name, nil)
-				if err != nil {
-					ghs.logger.Sugar().Errorf("error getting contributor count", zap.Error(err))
-				}
-
-				contribCount := 0
-				if len(contribs) > 0 {
-					contribCount = len(contribs)
-				}
-
-				ghs.logger.Sugar().Debugf("contributor count: %v for repo %v", contribCount, repo)
-
-				ghs.mb.RecordGitRepositoryContributorCountDataPoint(now, int64(contribCount), name)
-			}
-
-			count, err := getBranchCount(ctx, genClient, name, ghs.cfg.GitHubOrg)
-			if err != nil {
-				ghs.logger.Sugar().Errorf("error getting branch count", zap.Error(err))
-			}
-			ghs.logger.Sugar().Debugf("branch count: %v for repo %v", count.Repository.Refs.TotalCount, repo)
-
-			ghs.mb.RecordGitRepositoryBranchCountDataPoint(now, int64(count.Repository.Refs.TotalCount), name)
-
-			bp := getNumPages(float64(50), float64(count.Repository.Refs.TotalCount))
-			ghs.logger.Sugar().Debugf("branch pages: %v for repo %v", bp, repo)
-
-			for i := 0; i < bp; i++ {
-				r, err := getBranchData(ctx, genClient, name, ghs.cfg.GitHubOrg, 50, defaultBranch, branchCursor)
-				if err != nil {
-					ghs.logger.Sugar().Errorf("error getting branch data", zap.Error(err))
-				}
-
-				branches = append(branches, r.Repository.Refs.Nodes...)
-
-				branchCursor = &r.Repository.Refs.PageInfo.EndCursor
 
-			}
+		var wg1 sync.WaitGroup
+		var workers_per_op int = 3
 
-			for _, branch := range branches {
-				if branch.Name == defaultBranch || branch.Compare.BehindBy == 0 {
-					continue
-				}
+		chunkSize := (len(searchRepos) + workers_per_op - 1) / workers_per_op
```

Comment:
workers_per_op is snake case. Go convention is caml case. I'm surprised the go linter didn't catch that. What's the reasoning behind this function.

Reply:
Oh it's a buffer.